### PR TITLE
Add basic CI/CD pipeline

### PR DESCRIPTION
**Pull request by welpaolo**
This PR introduces basic CI/CD pipeline to format/linting code

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  workflow_call:
+  pull_request:
+  schedule:
+    - cron: '53 0 * * *' # Daily at 00:53 UTC
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
```
Comment (paulomach) [inline .github/workflows/ci.yaml:19]: common workflow
canonical/data-platform-workflows/.github/workflows/lint.yaml@v0.0.0

**Review by Batalex — DISMISSED** [review]
OK to merge as-is before using the common workflow, if that is blocking for #22

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (dragomirp) [reply .github/workflows/ci.yaml]: Added in https://github.com/canonical/data-platform-helpers/pull/18/commits/8715774b10637f1aacb45aaf03dfedb46af49910

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+[testenv:format]
+description = Apply coding style standards to code
+commands =
+    poetry install
+    poetry export -f requirements.txt -o requirements.txt --without-hashes
+
+    poetry install --only format
+    poetry run ruff check --fix {[vars]all_path}
+    poetry run black {[vars]all_path}
```
Comment (dragomirp) [inline tox.ini:37]: Do we want to use ruff for formatting?

File: data_platform_helpers/tests_helpers/__init__.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
 import logging
+from importlib.util import find_spec
 
-try:
-    import pytest_operator
-except ImportError as e:
```
Comment (dragomirp) [inline data_platform_helpers/tests_helpers/__init__.py:5]: Linting complains over the unused import, using find_spec to check if pytest_operator is installed

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,63 @@
+asttokens==2.4.1 ; python_version >= "3.10" and python_version < "4.0"
```
Comment (MiaAltieri) [inline requirements.txt:1]: is this needed? should it be added to gitignore?

File: tox.ini (hunk 2 above)
Comment (MiaAltieri) [reply tox.ini:37]: tbh I am not sure, I would be curious what others might think

File: requirements.txt (hunk 4 above)
Comment (dragomirp) [reply requirements.txt:1]: I think it's fine to ignore, will test locally and remove if nothing breaks.

Comment (dragomirp) [reply requirements.txt]: Doesn't look like we need to export, so removed in https://github.com/canonical/data-platform-helpers/pull/18/commits/80bb74bc28604b49f95e925cc76e40dadf4dccd0

**Review by MiaAltieri — DISMISSED** [review]

File: tox.ini (hunk 2 above)
Comment (Batalex) [reply tox.ini:36]: Since it's already there for linting, I am +1

Comment (paulomach) [reply tox.ini:36]: +1 to that. It's the trend everywhere on charming world

Comment (dragomirp) [reply tox.ini]: Added ruff formatting

**Review by Batalex — APPROVED** [review]

**Review by paulomach — APPROVED** [review]

**Pull request merged**